projects/lib/src/board/ncheckboard.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+    This file is part of Cute Chess.
+
+    Cute Chess is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Cute Chess is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Cute Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "ncheckboard.h"
+
+
+namespace Chess {
+
+/*! Maximum feasible value \a maxNCheck for nth-check limit. */
+static const int maxNCheck = 99;
+
+NCheckBoard::NCheckBoard(int n)
+	: StandardBoard(),
+	  m_checkLimit(qBound(1, n, maxNCheck))
+{
+}
+
+Board* NCheckBoard::copy() const
+{
+	return new NCheckBoard(*this);
+}
+
+QString NCheckBoard::variant() const
+{
+	return QString::number(checkLimit()) + "check";
+}
+
+QString NCheckBoard::defaultFenString() const
+{
+	return StandardBoard::defaultFenString() + " "
+	+ QString::number(checkLimit()) + "+"
+	+ QString::number(checkLimit());
+}
+
+void NCheckBoard::vInitialize()
+{
+	m_checksToWin[Side::White] = m_checksToWin[Side::Black] = checkLimit();
+	StandardBoard::vInitialize();
+}
+
+void NCheckBoard::updateCheckCounters(Side side, int d = -1)
+{
+	Q_ASSERT (m_checksToWin[Side::White] >=0);
+	Q_ASSERT (m_checksToWin[Side::Black] >=0);
+	Q_ASSERT (side != Side::NoSide);
+
+	if ( inCheck(side.opposite()) )
```

Comment:
Our convention is also not to pad parenthesis with spaces so this should be:

`if (inCheck(side.opposite()))`